packages/IntegrationTest/tests/lib/TestDefinition.tsx

```diff
@@ -12,63 +12,100 @@ import {AppRegistry, NativeModules, View} from 'react-native';
 const TestModule: {markTestPassed: (passed: boolean) => void} =
   NativeModules.TestModule;
 
-type TestFunction = (() => void) | (() => Promise<void>);
+export type TestFunction = (() => void) | (() => Promise<void>);
 
 /**
  * Define a new synchronus or asynchronus test function. The test may be failed
  * by throwing
  */
 export const functionTest = Object.assign(
   (name: string, fn: TestFunction) => {
-    const BoundTest: React.FunctionComponent = () => {
-      useEffect(() => {
-        (async () => {
-          try {
-            const testResult = fn();
-            if (testResult instanceof Promise) {
-              await testResult;
-            }
-          } catch (ex) {
-            // Need to force a redbox instead of YellowBox for unhandled
-            // promise rejection
-            ErrorUtils.getGlobalHandler()(ex);
-            return;
-          }
-
-          TestModule.markTestPassed(true);
-        })();
-      });
-
-      return <View />;
-    };
-
-    AppRegistry.registerComponent(name, () => BoundTest);
+    if (AppRegistry.getRunnable(name)) {
+      console.error(`Multiple components named "${name}" were registered`);
```

Comment:
>ultiple components [](start = 22, length = 18)

nit: maybe it would be cleaner to factor format string into small function for the 4 cases in this fil.

Reply:
actually ti seems the whole if-block could be shared between function and compontenttest...

---
In reply to: [492144514](https://github.com/microsoft/react-native-windows/pull/6045#discussion_r492144514) [](ancestors = 492144514)